src/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,95 @@
+export chow_ring, select
+
+function chow_ring(M::Matroid, extended=false)
+	Flats = flats(M)
+	sizeFlats = size(Flats)[1]
+	n = size(M.groundset)[1]
+	if(!is_loopless(M))
+		throw("Matroid has loops")
+	end
+	if(sizeFlats<2)
+		throw("matroid has to few flats")
+	end
+	proper_flats = Flats[2:sizeFlats-1]
+	var_names = [replace(string("x_",S), "["=>"{", "]"=>"}", ", "=>",") for S in proper_flats]# var names?
+	if(extended)
+		var_names = [var_names; [replace(string("h_",S), "["=>"{", "]"=>"}", ", "=>",") for S in [proper_flats;[Flats[sizeFlats]]]]]
+	end
+	s = size(var_names)
+
+	ring, vars = GradedPolynomialRing(QQ, var_names)
+	#ring, vars = PolynomialRing(ZZ, var_names) $which ring do we want here
+	I = linear_relations(ring, proper_flats, vars, M)
+	J = quadratic_relations(ring, proper_flats, vars)
+	Ex = []
+	if(extended)
+		Ex = relations_extended_ring(ring, proper_flats, vars)
+	end
+	chow_modulus = ideal(ring, vcat(I, J, Ex))
+	#chow_ring = ResidueRing(ring, chow_modulus)# is this a better choice?
+	chow_ring, projection = quo(ring, chow_modulus)
+	return chow_ring
+end
+
+function linear_relations(ring, proper_flats, vars, M)
+	alpha = 0
+	relations = Vector()
+	for i in M.groundset
+		poly = ring(0)
+		for index in findall(issubset([i],F) for F in proper_flats)
+			poly+= vars[index]
+		end
+		if(i==M.groundset[1])
+			alpha = poly
+		else
+			push!(relations, alpha-poly)
+		end
+	end
+	return relations
+end
+
+function quadratic_relations(ring, proper_flats, indeterminates)
+	relations = Vector()
+	s = size(proper_flats)[1]
+	for i in 1:s
+		F = proper_flats[i]
+		for j in 1:i-1
+			G = proper_flats[j]
+			if( !issubset(F,G) && !issubset(G,F) )
+				push!(relations, indeterminates[i]*indeterminates[j])
+			end
+		end
+	end
+	return relations
+end
+
+function relations_extended_ring(ring, proper_flats, vars)
+	relations = Vector()
```

Comment:
```suggestion
	relations = []
```
Better would be to specify a concrete type, though, if you can determine it in advance